octobot_trading/exchanges/util/parser/positions_parser.py

```diff
@@ -0,0 +1,342 @@
+                    f"allowed hedge values: {self.HEDGE_VALUES}",
+                )
+
+            self._try_to_find_and_parse_with_method(
+                PositioCols.POSITION_MODE.value,
+                self.MODE_KEY_NAMES,
+                parse_method=mode_found,
+                use_info_sub_dict=True,
+            )
+        else:
+            self._log_missing(
+                PositioCols.POSITION_MODE.value, "not implemented for this exchange"
+            )
+
+    def _parse_symbol(self):
+        # check is get_pair_from_exchange required? isn't ccxt already doing this?
+        self._try_to_find_and_parse_with_method(
+            PositioCols.SYMBOL.value,
+            [ExchangeCols.SYMBOL.value],
+            parse_method=self.exchange.get_pair_from_exchange,
+        )
+
+    def _parse_notional(self):
+        self._try_to_find_and_set_decimal(
+            PositioCols.NOTIONAL.value, [ExchangeCols.NOTIONAL.value]
+        )
+
+    def _parse_leverage(self):
+        self._try_to_find_and_set_decimal(
+            PositioCols.LEVERAGE.value, [ExchangeCols.LEVERAGE.value]
+        )
+
+    def _parse_realized_pnl(self):
+        def missing_realized():
+            pass
+        self._try_to_find_and_set_decimal(
+            PositioCols.REALISED_PNL.value,
+            [ExchangeCols.REALISED_PNL.value, BYBIT_REALIZED_PNL],
+            not_found_val=0,
+            use_info_sub_dict=True,
+            enable_log=False, # it's optional
+        )
+
+    def _parse_unrealized_pnl(self):
+        self._try_to_find_and_set_decimal(
+            PositioCols.UNREALIZED_PNL.value, [ExchangeCols.UNREALISED_PNL.value]
+        )
+
+    def _parse_status(self):
+        # todo improve - add LIQUIDATING, LIQUIDATED and ADL
+        size = self.formatted_record.get(PositioCols.SIZE.value)
+        try:
+            if size >= 0:
+                self.formatted_record[PositioCols.STATUS.value] = PositionStatus.OPEN
+            else:
+                self.formatted_record[PositioCols.STATUS.value] = PositionStatus.CLOSED
+        except Exception as e:
+            self._log_missing(
+                PositioCols.SIZE.value,
+                f"a valid size ({size or 'no size'}) is required to parse status, error: {e}",
+            )
+
+    def _parse_liquidation_price(self):
+        # todo is liquidation price different as bankruptcy price? merge?
+        self._try_to_find_and_set_decimal(
+            PositioCols.LIQUIDATION_PRICE.value,
+            [ExchangeCols.LIQUIDATION_PRICE.value, BYBIT_BANKRUPTCY_PRICE],
+        )
+
+    def _parse_value(self):
+        keys_to_find = [PositioCols.VALUE.value, BYBIT_VALUE]
+
+        def missing_value():
+            if (
+                mark_price := self.formatted_record.get(PositioCols.MARK_PRICE.value)
+            ) and (quantity := self.formatted_record.get(PositioCols.QUANTITY.value)):
+                return quantity / mark_price
```

Comment:
what if mark_price=0 ?

Reply:
mark price cant be 0, it'll raise with an error with report in the end